lib/vanagon/component/dsl.rb

```diff
@@ -163,36 +163,40 @@ def conflicts(pkgname, version = nil)
       #
       # @param service_file [String] path to the service file relative to the source
       # @param default_file [String] path to the default file relative to the source
-      # @param service_name [String] name of the service
-      # @param service_type [String] type of the service (network, application, system, etc)
-      # @param link_target [String] executable service file should be linked to
-      def install_service(service_file, default_file = nil, service_name = @component.name, service_type: nil, link_target: nil) # rubocop:disable Metrics/AbcSize
-        case @component.platform.servicetype
+      # @param options optional extra parameters
+      #        service_name [String] name of the service
+      #        service_type [String] type of the service (network, application, system, etc)
+      #        link_target [String] executable service file should be linked to
+      def install_service(service_file, default_file = nil, service_name = @component.name, **options) # rubocop:disable Metrics/AbcSize
+        servicetype = options[:init_system] || @component.platform.servicetype
```

Comment:
I find it a bit confusing to have this `servicetype` and the `service_type` option. Could one of them be called something else?

Reply:
The local variable could be renamed, but `servicetype` and `service_type` are already distinct fields in platform and component DSLs / definitely in use so those can't be renamed.

Reply:
updated.